addons/web/static/tests/views/fields/reference_field_tests.js

```diff
@@ -964,6 +964,54 @@ QUnit.module("Fields", (hooks) => {
         }
     );
 
+    QUnit.test(
+        "Change model field of a ReferenceField then select an invalid value (tree list in form view)",
+        async function (assert) {
+            serverData.models.turtle.records[0].partner_ids = [1];
+            serverData.models.partner.records[0].reference = "product,41";
+            serverData.models.partner.records[0].model_id = 20;
+
+            await makeView({
+                type: "form",
+                resModel: "turtle",
+                resId: 1,
+                serverData,
+                arch: `
+                    <form>
+                        <field name="partner_ids">
+                            <tree editable="bottom">
+                                <field name="name" />
+                                <field name="model_id"/>
+                                <field name="reference" required="true" options="{'model_field': 'model_id'}" class="reference_field" />
+                            </tree>
+                        </field>
+                   </form>`,
+            });
+            assert.strictEqual(target.querySelector(".reference_field").textContent, "xpad");
+            assert.strictEqual(target.querySelector(".o_list_many2one").textContent, "Product");
+
+            await click(target, ".o_list_table td.o_list_many2one");
+            await click(target, ".o_list_table .o_list_many2one input");
+            //make sure the options are there
+            assert.ok(1 < [...target.querySelectorAll(".o_list_table .o_list_many2one ul li")].length);
```

Comment:
Don't we know the size? I don't like testing "greater than" or "less than", it could be 3, it could be 6745.

Reply:
(Well testing against another variable is not a lot better as the "size" could be 0, but well, not a huge deal ^^).

Reply:
I can just remove the line realistically. The idea was probably to ensure I'm actually changing the model, which wouldn't be possible with only one option using the keyboard. But since now I explicitly select "partner" and we set the reference field to 'product' at the start it's not useful anymore